[Pages] Functions binding revision (#7390)
cloudflare/cloudflare-docs: content/pages/platform/functions/bindings.md

# Bindings

> A [binding](/workers/platform/bindings/) enables your Pages Functions to interact with resources on the Cloudflare developer platform. Use bindings to integrate your Pages Functions with Cloudflare resources like [KV](/workers/learning/how-kv-works/), [Durable Objects](/workers/learning/using-durable-objects/), [R2](https://developers.cloudflare.com/r2/), and [D1](https://blog.cloudflare.com/introducing-d1/). You can set bindings for both production and preview environments. 

Reviewer: ```suggestion
A [binding](/workers/platform/bindings/) enables your Pages Functions to interact with resources on the Cloudflare developer platform. Use bindings to integrate your Pages Functions with Cloudflare resources like [KV](/workers/learning/how-kv-works/), [Durable Objects](/workers/learning/using-durable-objects/), [R2](/r2/), and [D1](/d1/). You can set bindings for both production and preview environments. 
```

This guide will instruct you on configuring a binding for your Pages Function. You must already have a resource set up to continue.

## KV namespaces

> [Workers KV](https://www.cloudflare.com/products/workers-kv/) is Cloudflare's key-value storage solution. To bind your KV namespace to your Pages Function:

Reviewer: ```suggestion
[Workers KV](/workers/wrangler/workers-kv/) is Cloudflare's key-value storage solution. To bind your KV namespace to your Pages Function:
```

### Interact with your KV namespaces locally

> While developing locally, interact with your KV namespace by adding `-k <BINDING_NAME>` or `--kv <BINDING_NAME>` to your run command. For example, if your namespace is bound to `TODO_LIST`, access the KV namespace in your local dev by running `npx wrangler pages dev <OUTPUT_DIR> --kv TODO_LIST`. The data from this namespace can be accessed using `context.env.TODO_LIST`.

Reviewer 2: Consistency with the rest

```suggestion
While developing locally, interact with your KV namespace by adding `-k <BINDING_NAME>` or `--kv=<BINDING_NAME>` to your run command. For example, if your namespace is bound to `TODO_LIST`, access the KV namespace in your local dev by running `npx wrangler pages dev <OUTPUT_DIR> --kv=TODO_LIST`. The data from this namespace can be accessed using `context.env.TODO_LIST`.
```

## R2 buckets

> [R2](https://www.cloudflare.com/products/r2/) is Cloudflare's blob storage solution that allows developers to store large amounts of unstructured data without the egress fees. To bind your R2 bucket to your Pages Function:

Reviewer: ```suggestion
[R2](/r2/) is Cloudflare's blob storage solution that allows developers to store large amounts of unstructured data without the egress fees. To bind your R2 bucket to your Pages Function:
```

### Interact with your R2 buckets locally

> While developing locally, interact with an R2 bucket by adding `--r2 <BINDING_NAME>` to your run command. For example, if your bucket is bound to `BUCKET`, access this bucket in local dev by running `npx wrangler pages dev <OUTPUT_DIR> --r2=BUCKET`. Interact with this binding by using `context.env` (for example, `context.env.BUCKET`).

Reviewer 2: Either work but yes, we should switch to `--r2=<BINDING_NAME>` to keep consistent with the rest of this

### Interact with your D1 databases locally

> While developing locally, interact with a D1 database by adding `--d1 <BINDING_NAME>` to your run command. For example, if your database is bound to `NORTHWIND_DB`, access this database in local dev by running `npx wrangler pages dev <OUTPUT_DIR> --d1=NORTHWIND_DB`. Interact with this binding by `using context.env` (for example, `context.env.NORTHWIND_DB`).

Author: same thing @WalshyDev

Reviewer 2: Same answer :)

## Service bindings

> Service bindings enable you to call a Worker from within your Pages Function. To add a service binding to your Pages Function:

Reviewer: ```suggestion
[Service bindings](/workers/runtime-apis/service-bindings/) enable you to call a Worker from within your Pages Function. To add a service binding to your Pages Function:
```

{{</tab>}}
{{</tabs>}}

> ### Interact with your service binding locally

Reviewer 2: This should be capital no? Service (Worker) is a name of something. We don't use the name Service a lot but it is a name like Durable Objects.

> While developing locally, interact with a service by adding `--service <BINDING_NAME>=<Worker>` to your run command. For example, if your service is bound to `SERVICE`, access this service in local dev by running `npx wrangler pages dev <OUTPUT_DIR> --service=SERVICE=my-worker`. You will need to also have the `my-worker` Worker running in `wrangler pages dev --local`. Interact with this binding by using context.env (for example, `context.env.SERVICE`).

Author: same thing here @WalshyDev

> While developing locally, interact with a service by adding `--service <BINDING_NAME>=<Worker>` to your run command. For example, if your service is bound to `SERVICE`, access this service in local dev by running `npx wrangler pages dev <OUTPUT_DIR> --service=SERVICE=my-worker`. You will need to also have the `my-worker` Worker running in `wrangler pages dev --local`. Interact with this binding by using `context.env` (for example, `context.env.SERVICE`).

Author: same thing here between second and first sentence commands not matching @WalshyDev

Reviewer 2: ye ye same thing

### Interact with your environment variables locally

When developing locally, add environment variables by creating a `.dev.vars` file in the root directory of your project. Then add the following code snippet to `.dev.vars`:

```
---

> filename:  `.dev.vars`

Author: @WalshyDev

Reviewer 2: Yep that's correct

---
ENVIRONMENT=development
```

## Secrets

Secrets are environment variables that are encrypted and not visible once set. They are used for storing sensitive information like API keys, and auth tokens.

To add secrets to your Pages project:

1. Log in to the [Cloudflare dashboard](https://dash.cloudflare.com).
2. In **Account Home**, select **Pages**.
3. In your Pages project, select **Settings** > **Environment variables**.
4. Selecting **Add variables** under **Production** and/or **Preview**.

> 5. Select **Encrypt** to create your secret.

Author: added this because it looked incomplete please check @WalshyDev

Reviewer 2: Haha oops, yeah this looks good. Thank you ❤️
